[BUG] @Getter(lazy=true) Compilation error (no suitable method found)

**Describe the bug**
The code below gives this compilation error:
```
11:55:45: Executing ':Main.main()'...


> Task :compileJava FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
1 actionable task: 1 executed
W:\Projects\tests\Lombok-Bug\src\main\java\org\example\Main.java:15: error: no suitable method found for collect(Collector<String,CAP#1,Map<Integer,Object>>)
    @Getter(lazy = true)
    ^
    method Stream.<R#1>collect(Supplier<R#1>,BiConsumer<R#1,? super String>,BiConsumer<R#1,R#1>) is not applicable
      (cannot infer type-variable(s) R#1
        (actual and formal argument lists differ in length))
    method Stream.<R#2,A>collect(Collector<? super String,A,R#2>) is not applicable
      (cannot infer type-variable(s) R#2,A
        (argument mismatch; Duration cannot be converted to Collector<? super String,A,R#2>))
  where R#1,T,R#2,A are type-variables:
    R#1 extends Object declared in method <R#1>collect(Supplier<R#1>,BiConsumer<R#1,? super T>,BiConsumer<R#1,R#1>)
    T extends Object declared in interface Stream
    R#2 extends Object declared in method <R#2,A>collect(Collector<? super T,A,R#2>)
    A extends Object declared in method <R#2,A>collect(Collector<? super T,A,R#2>)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object from capture of ?
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 704ms
11:55:46: Execution finished ':Main.main()'.
```

**To Reproduce**
**`Main.java`**
```java
package org.example;

import lombok.Getter;
import reactor.core.publisher.Mono;

import java.time.Duration;
import java.util.Arrays;
import java.util.Map;
import java.util.stream.Collectors;

public class Main {


    @Getter(lazy = true)
    private final Mono<Map<Integer, String>> mono =
            Mono.just(new String[]{"a", "b", "c", "d"})
                    .map(Arrays::asList)
                    .map(list -> list.stream().collect(Collectors.toMap(String::hashCode, v -> v)))
                    .cache(a -> Duration.ofSeconds(10),
                            v -> Duration.ofSeconds(10),
                            () -> Duration.ofSeconds(10)
                    );

/* Similar code that works as expected

    @Getter(lazy = true)
    private final Mono<Map<Integer, String>> mono =
            Mono.just(new String[]{"a", "b", "c", "d"})
                    .map(Arrays::asList)
                    .map(list -> list.stream().collect(Collectors.toMap(String::hashCode, v -> v)))
                    .cache(a -> Duration.ofSeconds(10),
                            v -> Duration.ofSeconds(10),
                            () -> Duration.ZERO   // <--------The only difference is in this line
                    );


    @Getter(lazy = true)
    private final Mono<Map<Integer, String>> mono =
            Mono.just(new String[]{"a", "b", "c", "d"})
                    .map(Arrays::asList)
                    .map(list -> list.stream().collect(Collectors.toMap(String::hashCode, v -> v)));

*/


    public static void main(String[] args) {
        new Main().getMono().subscribe(v -> System.out.println(v));
    }
}
```

**`build.gradle`**
```
plugins {
    id 'java'
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    compileOnly 'org.projectlombok:lombok:1.18.24'
    annotationProcessor 'org.projectlombok:lombok:1.18.24'

    implementation 'io.projectreactor:reactor-core:3.5.1'
}
```

**Expected behavior**
Successful compilation. Code should run as expected.

**Version info:**
 - Lombok version: `1.18.24`
 - JDK: `17.0.1`, `17.0.5`.
 - IDE: Intellij Idea `2022.3`
 - Gradle: `7.5.1`

--- a/src/core/lombok/javac/JavacAugments.java
+++ b/src/core/lombok/javac/JavacAugments.java
@@ -34,4 +34,5 @@ public final class JavacAugments {
 	public static final FieldAugment<JCTree, Boolean> JCTree_handled = FieldAugment.augment(JCTree.class, boolean.class, "lombok$handled");
 	public static final FieldAugment<JCTree, JCTree> JCTree_generatedNode = FieldAugment.circularSafeAugment(JCTree.class, JCTree.class, "lombok$generatedNode");
 	public static final FieldAugment<JCImport, Boolean> JCImport_deletable = FieldAugment.circularSafeAugment(JCImport.class, Boolean.class, "lombok$deletable");
+	public static final FieldAugment<JCTree, Boolean> JCTree_keepPosition = FieldAugment.augment(JCTree.class, boolean.class, "lombok$keepPosition");
 }

--- a/src/core/lombok/javac/handlers/HandleGetter.java
+++ b/src/core/lombok/javac/handlers/HandleGetter.java
@@ -23,6 +23,7 @@ package lombok.javac.handlers;
 
 import static lombok.core.handlers.HandlerUtil.*;
 import static lombok.javac.Javac.*;
+import static lombok.javac.JavacAugments.JCTree_keepPosition;
 import static lombok.javac.JavacTreeMaker.TypeTag.*;
 import static lombok.javac.handlers.JavacHandlerUtil.*;
 
@@ -47,7 +48,6 @@ import lombok.javac.JavacTreeMaker.TypeTag;
 import lombok.spi.Provides;
 
 import com.sun.tools.javac.code.Flags;
-import com.sun.tools.javac.tree.JCTree;
 import com.sun.tools.javac.tree.JCTree.JCAnnotation;
 import com.sun.tools.javac.tree.JCTree.JCBinary;
 import com.sun.tools.javac.tree.JCTree.JCBlock;
@@ -229,18 +229,9 @@ public class HandleGetter extends JavacAnnotationHandler<Getter> {
 		boolean makeFinal = shouldMakeFinal(field, accessors);
 		
 		List<JCStatement> statements;
-		JCTree toClearOfMarkers = null;
-		int[] methodArgPos = null;
 		boolean addSuppressWarningsUnchecked = false;
 		if (lazy && !inNetbeansEditor(field)) {
-			toClearOfMarkers = fieldNode.init;
-			if (toClearOfMarkers instanceof JCMethodInvocation) {
-				List<JCExpression> args = ((JCMethodInvocation) toClearOfMarkers).args;
-				methodArgPos = new int[args.length()];
-				for (int i = 0; i < methodArgPos.length; i++) {
-					methodArgPos[i] = args.get(i).pos;
-				}
-			}
+			JCTree_keepPosition.set(fieldNode.init, true);
 			statements = createLazyGetterBody(treeMaker, field, source);
 			addSuppressWarningsUnchecked = LombokOptionsFactory.getDelombokOptions(field.getContext()).getFormatPreferences().generateSuppressWarnings();
 		} else {
@@ -268,12 +259,6 @@ public class HandleGetter extends JavacAnnotationHandler<Getter> {
 		JCMethodDecl decl = recursiveSetGeneratedBy(treeMaker.MethodDef(treeMaker.Modifiers(access, annsOnMethod), methodName, methodType,
 			methodGenericParams, parameters, throwsClauses, methodBody, annotationMethodDefaultValue), source);
 		
-		if (toClearOfMarkers != null) recursiveSetGeneratedBy(toClearOfMarkers, null);
-		if (methodArgPos != null) {
-			for (int i = 0; i < methodArgPos.length; i++) {
-				((JCMethodInvocation) toClearOfMarkers).args.get(i).pos = methodArgPos[i];
-			}
-		}
 		decl.mods.annotations = decl.mods.annotations.appendList(delegates);
 		if (addSuppressWarningsUnchecked) {
 			ListBuffer<JCExpression> suppressions = new ListBuffer<JCExpression>();

--- a/src/core/lombok/javac/handlers/JavacHandlerUtil.java
+++ b/src/core/lombok/javac/handlers/JavacHandlerUtil.java
@@ -25,6 +25,7 @@ import static com.sun.tools.javac.code.Flags.GENERATEDCONSTR;
 import static lombok.core.handlers.HandlerUtil.*;
 import static lombok.javac.Javac.*;
 import static lombok.javac.JavacAugments.JCTree_generatedNode;
+import static lombok.javac.JavacAugments.JCTree_keepPosition;
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Constructor;
@@ -126,6 +127,7 @@ public class JavacHandlerUtil {
 		
 		@Override public void scan(JCTree tree) {
 			if (tree == null) return;
+			if (JCTree_keepPosition.get(tree)) return;
 			setGeneratedBy(tree, source);
 			super.scan(tree);
 		}
@@ -173,10 +175,11 @@ public class JavacHandlerUtil {
 		}
 		JCTree_generatedNode.set(node, sourceNode.get());
 		
-		if (!inNetbeansEditor(sourceNode.getContext()) || isParameter(node)) {
-			node.pos = sourceNode.getStartPos();
-			storeEnd(node, sourceNode.getEndPosition(), (JCCompilationUnit) sourceNode.top().get());
-		}
+		if (JCTree_keepPosition.get(node)) return node;
+		if (inNetbeansEditor(sourceNode.getContext()) && !isParameter(node)) return node;
+		
+		node.pos = sourceNode.getStartPos();
+		storeEnd(node, sourceNode.getEndPosition(), (JCCompilationUnit) sourceNode.top().get());
 		return node;
 	}
 

--- a/test/transform/resource/after-delombok/GetterLazyArguments.java
+++ b/test/transform/resource/after-delombok/GetterLazyArguments.java
@@ -17,6 +17,7 @@ class GetterLazyArguments {
 	private final java.util.concurrent.atomic.AtomicReference<java.lang.Object> field3 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
 	private final java.util.concurrent.atomic.AtomicReference<java.lang.Object> field4 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
 	private final java.util.concurrent.atomic.AtomicReference<java.lang.Object> field5 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
+	private final java.util.concurrent.atomic.AtomicReference<java.lang.Object> field6 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
 
 	@java.lang.SuppressWarnings({"all", "unchecked"})
 	public String getField1() {
@@ -98,5 +99,21 @@ class GetterLazyArguments {
 			}
 		}
 		return (String) (value == this.field5 ? null : value);
+	}
+
+	@java.lang.SuppressWarnings({"all", "unchecked"})
+	public String getField6() {
+		java.lang.Object value = this.field6.get();
+		if (value == null) {
+			synchronized (this.field6) {
+				value = this.field6.get();
+				if (value == null) {
+					final String actualValue = true ? stringInt(true ? "a" : "b", true ? 1 : 0) : "";
+					value = actualValue == null ? this.field6 : actualValue;
+					this.field6.set(value);
+				}
+			}
+		}
+		return (String) (value == this.field6 ? null : value);
 	}
 }

--- a/test/transform/resource/after-delombok/GetterLazyErrorPosition.java
+++ b/test/transform/resource/after-delombok/GetterLazyErrorPosition.java
@@ -1,0 +1,19 @@
+class GetterLazyErrorPosition {
+	private final java.util.concurrent.atomic.AtomicReference<java.lang.Object> field = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
+
+	@java.lang.SuppressWarnings({"all", "unchecked"})
+	public String getField() {
+		java.lang.Object value = this.field.get();
+		if (value == null) {
+			synchronized (this.field) {
+				value = this.field.get();
+				if (value == null) {
+					final String actualValue = true ? "" : new ErrorPosition();
+					value = actualValue == null ? this.field : actualValue;
+					this.field.set(value);
+				}
+			}
+		}
+		return (String) (value == this.field ? null : value);
+	}
+}

--- a/test/transform/resource/after-ecj/GetterLazyArguments.java
+++ b/test/transform/resource/after-ecj/GetterLazyArguments.java
@@ -4,6 +4,7 @@ class GetterLazyArguments {
   private final @lombok.Getter(lazy = true) java.util.concurrent.atomic.AtomicReference<java.lang.Object> field3 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
   private final @lombok.Getter(lazy = true) java.util.concurrent.atomic.AtomicReference<java.lang.Object> field4 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
   private final @lombok.Getter(lazy = true) java.util.concurrent.atomic.AtomicReference<java.lang.Object> field5 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
+  private final @lombok.Getter(lazy = true) java.util.concurrent.atomic.AtomicReference<java.lang.Object> field6 = new java.util.concurrent.atomic.AtomicReference<java.lang.Object>();
   GetterLazyArguments() {
     super();
   }
@@ -102,5 +103,22 @@ class GetterLazyArguments {
             }
         }
     return (String) ((value == this.field5) ? null : value);
+  }
+  public @java.lang.SuppressWarnings({"all", "unchecked"}) String getField6() {
+    java.lang.Object value = this.field6.get();
+    if ((value == null))
+        {
+          synchronized (this.field6)
+            {
+              value = this.field6.get();
+              if ((value == null))
+                  {
+                    final String actualValue = (true ? stringInt((true ? "a" : "b"), (true ? 1 : 0)) : "");
+                    value = ((actualValue == null) ? this.field6 : actualValue);
+                    this.field6.set(value);
+                  }
+            }
+        }
+    return (String) ((value == this.field6) ? null : value);
   }
 }

--- a/test/transform/resource/before/GetterLazyArguments.java
+++ b/test/transform/resource/before/GetterLazyArguments.java
@@ -18,4 +18,7 @@ class GetterLazyArguments {
 	
 	@lombok.Getter(lazy=true)
 	private final String field5 = stringRunnable(("a"), () -> { });
+	
+	@lombok.Getter(lazy=true)
+	private final String field6 = true ? stringInt(true ? "a" : "b", true ? 1 : 0) : "";
 }

--- a/test/transform/resource/before/GetterLazyErrorPosition.java
+++ b/test/transform/resource/before/GetterLazyErrorPosition.java
@@ -1,0 +1,8 @@
+//platform javac: positions in eclipse are hard...
+class GetterLazyErrorPosition {
+	@lombok.Getter(lazy=true)
+	private final String field = 
+		true ? 
+		"" : 
+		new ErrorPosition();
+}

--- a/test/transform/resource/messages-delombok/GetterLazyErrorPosition.java.messages
+++ b/test/transform/resource/messages-delombok/GetterLazyErrorPosition.java.messages
@@ -1,0 +1,1 @@
+6 cannot find symbol symbol: class ErrorPosition location: class GetterLazyErrorPosition

--- a/test/transform/resource/messages-delombok/GetterLazyErrorPosition.java.messages
+++ b/test/transform/resource/messages-delombok/GetterLazyErrorPosition.java.messages
@@ -1,1 +1,1 @@
-6 cannot find symbol symbol: class ErrorPosition location: class GetterLazyErrorPosition
+7 cannot find symbol

--- a/test/transform/resource/messages-idempotent/GetterLazyErrorPosition.java.messages
+++ b/test/transform/resource/messages-idempotent/GetterLazyErrorPosition.java.messages
@@ -1,0 +1,1 @@
+11 cannot find symbol symbol: class ErrorPosition location: class GetterLazyErrorPosition

--- a/test/transform/resource/messages-idempotent/GetterLazyErrorPosition.java.messages
+++ b/test/transform/resource/messages-idempotent/GetterLazyErrorPosition.java.messages
@@ -1,1 +1,1 @@
-11 cannot find symbol symbol: class ErrorPosition location: class GetterLazyErrorPosition
+11 cannot find symbol

